### DISCUSSION TITLE
Thread support?

### DISCUSSION THREAD

**Discussion by jankornblum** [HomeSpan/HomeSpan#481] [General]
Hello!

Does or will HomeSpan support thread? Seems there is a new ESP32-C6 available now supporting thread.

And, will Ethernet be supported too? This would be really great especially for building PoE based devices!

**Comment by HomeSpan** [comment]
HomeSpan does not support thread and only works with Espressif's Arduino-ESP32 library, which does not yet support the ESP32-C6.

As for Ethernet, I tried adding support for this a while back but there is a problem with the underlying UDP protocol on the ESP32 which prevented MDNS from working properly.  I may try again in the future once Espressif updates the Arduino-ESP32 library to use the Espressif IDF version 5 (the Arduino library is only on version 4.4.3).

**Reply by jankornblum** [reply]
Thx a lot! So when the Arduino Library will support ESP32-C6, will HomeSpan automatically support thread too? Or must this be implemented to HomeSpan too?

**Reply by HomeSpan** [reply]
I've been reviewing some of the Thread documentation and it does not look like I'l be able to get this to work with HomeSpan (at least at this time) since Apple has not released any specs for non-commercial use of Thread with HomeKit.  At present it seems Thread is restricted to be used by only commercial Accessories with HomeKit MiFi licenses.

**Reply by jankornblum** [reply]
@HomeSpan Thx! That would be very regrettable if apple wouldn’t allow uncommercial usage.

**Reply by jehnes** [reply]
I tried using Homespun with the ESP32 library version 3.0.4, which supports the ESP32-C6, but I get the following error:
#error ERROR: HOMESPAN REQUIRES VERSION 2 OF THE ARDUINO ESP32 LIBRARY.  HOMESPAN IS NOT COMPATIBLE WITH VERSION 1 OR VERSION 3

When I downgrade to version 2.0.17 (latest version2), ESP32-C6 is not supported. So the issue seems to be that Homespun is not up to date. Of course without being able to build the application I cannot test if Apple has any other issues with thread and HomeKit. But since they demonstrated it using a different library at WWDC, I guess that should not be an issue.

**Comment by HomeSpan** [comment]
HomeSpan 2.0.0-rc.1 has just been released.  It is designed for version 3 of the Arduino-ESP32 Board Manager (requires 3.0.2 or greater).  However, though the Arduino-ESP32 library supports Thread, HomeSpan does not make any use of it (no plans to do so).
